k2/python/k2/fsa.py

```diff
@@ -391,6 +391,8 @@ def __setattr__(self, name: str, value: Any) -> None:
 
         if isinstance(value, torch.Tensor):
             assert value.shape[0] == self.arcs.values().shape[0]
+            # See https://github.com/k2-fsa/k2/issues/1120
+            assert value.device == self.device
```

Comment:
Please also change line 417 and add tests for your changes.

Reply:
See https://github.com/k2-fsa/k2/pull/1123

Please always add tests for your changes in the future if possible.